contracts/test/v0.8/dev/UpkeepTranscoder3_0.test.ts

```diff
@@ -0,0 +1,449 @@
+            linkToken.address,
+            linkEthFeed.address,
+            gasPriceFeed.address,
+          )
+
+        let registry20 = await keeperRegistryFactory20
+          .connect(owner)
+          .deploy(registryLogic.address)
+
+        // deploys a registry, setups of initial configuration, registers an upkeep
+        let keeper1 = personas.Carol
+        let keeper2 = personas.Eddy
+        let keeper3 = personas.Nancy
+        let keeper4 = personas.Norbert
+        let keeper5 = personas.Nick
+        let payee1 = personas.Nelly
+        let payee2 = personas.Norbert
+        let payee3 = personas.Nick
+        let payee4 = personas.Eddy
+        let payee5 = personas.Carol
+        // signers
+        let signer1 = new ethers.Wallet(
+          '0x7777777000000000000000000000000000000000000000000000000000000001',
+        )
+        let signer2 = new ethers.Wallet(
+          '0x7777777000000000000000000000000000000000000000000000000000000002',
+        )
+        let signer3 = new ethers.Wallet(
+          '0x7777777000000000000000000000000000000000000000000000000000000003',
+        )
+        let signer4 = new ethers.Wallet(
+          '0x7777777000000000000000000000000000000000000000000000000000000004',
+        )
+        let signer5 = new ethers.Wallet(
+          '0x7777777000000000000000000000000000000000000000000000000000000005',
+        )
+
+        let keeperAddresses = [
+          await keeper1.getAddress(),
+          await keeper2.getAddress(),
+          await keeper3.getAddress(),
+          await keeper4.getAddress(),
+          await keeper5.getAddress(),
+        ]
+        let payees = [
+          await payee1.getAddress(),
+          await payee2.getAddress(),
+          await payee3.getAddress(),
+          await payee4.getAddress(),
+          await payee5.getAddress(),
+        ]
+        let signers = [signer1, signer2, signer3, signer4, signer5]
+
+        let signerAddresses = []
+        for (let signer of signers) {
+          signerAddresses.push(await signer.getAddress())
+        }
+
+        let f = 1
+        let offchainVersion = 1
+        let offchainBytes = '0x'
+
+        await registry20
+          .connect(owner)
+          .setConfig(
+            signerAddresses,
+            keeperAddresses,
+            f,
+            encodeConfig(config),
+            offchainVersion,
+            offchainBytes,
+          )
+        await registry20.connect(owner).setPayees(payees)
+        await linkToken
+          .connect(owner)
+          .approve(registry12.address, toWei('1000'))
+        await registry12.connect(owner).addFunds(id, toWei('1000'))
+
+        // set outgoing permission to registry 2_0 and incoming permission for registry 1_2
+        await registry12.setPeerRegistryMigrationPermission(
+          registry20.address,
+          1,
+        )
+        await registry20.setPeerRegistryMigrationPermission(
+          registry12.address,
+          2,
+        )
```

Comment:
just a thought on code cleanliness.. might be nice to separate some of this setup stuff from the actual test. Could do...

```ts
it('migrates upkeeps from one registry to another', () => {
  registry12, registry20, registry30 = deployRegistries()
  // actual test...
})

function deployRegistries() {
  // setup code no one cares about
}
```